templates/base/src/localization/useLocalizsedContent.js

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from 'react';
+import fetchWrapper from '../utils/serviceUtils/fetch';
+
+const languageStorageKey = "language";
+var cache={};
+
+// Default base path
+const basePath = "http://localhost:3000";
+
+const getLanguageFromLocalStorage = (nullCheck) => {
+  const language = localStorage.getItem(languageStorageKey);
+   return nullCheck(language);
+}
+
+const nullCheck = (variable) => {
+  if(!variable){
+    // Default language
+    return "en-us";
+  }
+  else{
+    return variable;
+  }
+}
+
+const setLanguageToLocalStorage = (language) => {
+  localStorage.setItem(languageKey, language);
+}
+
+export default function useLocalisedContent(url, fetchOptions){
+
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState();
+
+  //Logic for handling multi locale requests
+  const language = getLanguageFromLocalStorage(nullCheck);
```

Comment:
Why is this nullCheck function passed as Argument? It's available in same scope. It another thing that its not needed.